### PR TITLE
tests: Update GoogleTest to latest

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Use the main branch per GoogleTest's recommendation.

### CODE REVIEW

File: scripts/known_good.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
               "-Dgtest_force_shared_crt=ON",
               "-DBUILD_SHARED_LIBS=OFF"
         ],
-        "commit": "release-1.8.1",
+        "commit": "main",
```
Comment (jeremyg-lunarg) [inline scripts/known_good.json:64]: do we want to use main or just a newer release tag?

**Review by jeremyg-lunarg — COMMENTED** [review]
+eleventy for updating.

File: scripts/known_good.json (hunk 1 above)
Comment (ncesario-lunarg) [reply scripts/known_good.json:64]: I first went looking for a tag and noticed the last one is [nearly a year old](https://github.com/google/googletest/releases/tag/release-1.11.0). I then noticed this in the docs: https://github.com/google/googletest#live-at-head. We could run into failures one way or the other down the road, and I'm personally fine with this particular lib living at head, but I'm also fine locking this at an arbitrary commit if you prefer.

Comment (jeremyg-lunarg) [reply scripts/known_good.json:64]: I didn't realize the last tag was so old. works for me.

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**